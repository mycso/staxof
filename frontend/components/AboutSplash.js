import Link from 'next/link';
import styled from 'styled-components';

import { ScrollTo } from "react-scroll-to"
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { 
    faArrowDown,
} from '@fortawesome/free-solid-svg-icons';

import { 
    faFacebook,
    faTwitter,
    faInstagram
} from '@fortawesome/free-brands-svg-icons';

import SplashImage from './styles/SplashImageStyles';
import SubHeader from '../components/SubHeader';

const StyledAboutSplash = styled.div`

	.hi-five {
		background-image: url('static/hi-five.jpeg');
		background-size: cover;
		width: 100%;
		height: 100%;
		background-position: top;
	}
	
`;

const AboutSplash = () => (
	<SplashImage>
		<StyledAboutSplash>
			<SubHeader />
			<div>
			  <div id="carouselExampleIndicators" className="carousel slide" data-ride="carousel">
			    <ol className="carousel-indicators d-none">
			      <li data-target="#carouselExampleIndicators" data-slide-to="0" className="active"></li>
			      <li data-target="#carouselExampleIndicators" data-slide-to="1"></li>
			      <li data-target="#carouselExampleIndicators" data-slide-to="2"></li>
			    </ol>
			    <div className="social-button-main">
			      <div className="col-xs-12 col-sm-12 col-md-12">
			        <ul className="list-unstyled list-inline social text-center">
			          <li className="list-inline-item"><a href="https://www.facebook.com/staxofstories/" target="_blank"><FontAwesomeIcon icon={faFacebook} /></a></li>
			          <li className="list-inline-item"><a href="#"><FontAwesomeIcon icon={faTwitter} /></a></li>
			          <li className="list-inline-item"><a href="https://www.instagram.com/staxofstories/" target="_blank"><FontAwesomeIcon icon={faInstagram} /></a></li>
			        </ul>
			      </div>
			    </div> 
			    <div className="carousel-inner" role="listbox">
			      <div className="carousel-item active">
			      	<div className="hi-five"></div>
			        <div className="carousel-caption heading d-md-block"> 
			          <h2 className="display-4">About Us!</h2>
			          <h3 className="display-5">Discover who we really are.</h3>
			        </div>
			      </div>
			    </div>
			    <a className="carousel-control-prev d-none" href="#carouselExampleIndicators" role="button" data-slide="prev">
			      <span className="carousel-control-prev-icon" aria-hidden="true"></span>
			      <span className="sr-only">Previous</span>
			    </a>
			    <a className="carousel-control-next d-none" href="#carouselExampleIndicators" role="button" data-slide="next">
			      <span className="carousel-control-next-icon" aria-hidden="true"></span>
			      <span className="sr-only">Next</span>
			    </a>
			  </div>
		  	</div>
		  </StyledAboutSplash>
	  </SplashImage>
);

export default AboutSplash;