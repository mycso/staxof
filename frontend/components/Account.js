import NavLink from './Link'
import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { format } from 'date-fns';
import Head from 'next/head';
import gql from 'graphql-tag';
import formatMoney from '../lib/formatMoney';
import Error from './ErrorMessage';
import styled from 'styled-components';
import OrderStyles from './styles/OrderStyles';
import SubHeader from '../components/SubHeader';
import DeleteUser from './DeleteUser'; 

const SINGLE_ORDER_QUERY = gql`
	query SINGLE_ORDER_QUERY($id: ID!) {
		order(id: $id) {
			id
			charge
			total
			createdAt
			user {
				id
			}
			items {
				id
				title
				description
				price
				image
				quantity
			}
		}
	}
`;

const CURRENT_USER_QUERY = gql`
	query {
		me {
			id
			email
			name
			permissions
			cart {
				id
				quantity
				item {
					id
					price
					image
					title
					description
				}
			}

			order {
				id
				charge
				total
			}
		}
	}
`;

const StyledLinked = styled.div`

  	a {
    	color: ${props => props.theme.navyBlue};
  	}

  	a:hover {
    	color: ${props => props.theme.navyBlue};
    	text-decoration: none;
  	}

`;

class Account extends React.Component {
	static propTypes = {
		id: PropTypes.string.isRequired,
	};
	render() {
		return (
			<StyledLinked>
				<Query query={CURRENT_USER_QUERY} variables={{ id: this.props.id }}>
					{({ data, error, loading }) => {
						if (error) return <Error error={error} />;
						if (loading) return <p>Loading</p>;
						const user = data.me;
						const me = data ? data.me : null;

						return (
							
								<div>
									<SubHeader />
									<OrderStyles data-test="user">
										<p>
											<span>User ID:</span> 
											<span>{user.id}</span>
										</p>
										<p>
											<span>Email:</span>
											<span>{user.email}</span>
										</p>

										<p>
											<span>Account Action:</span>
											<span>
									        	<a><DeleteUser id={user.id}>Delete Account</DeleteUser></a>
									        </span>
										</p>
											{me && (
										      	<>
													<p>
														<span>Orders:</span>
												      	<span>
												      		<NavLink activeClassName='active' prefetch="true" href="/orders">
												        		<u><a href="">Purchased Orders</a></u>
												        	</NavLink>
													      	
												        </span>
											        </p>

											        {me.permissions.includes("ADMIN") && (
										     	  	<>
												        <p>
												        	<span>All User Permissions:</span>
													      	<span>
														        <NavLink activeClassName='active' prefetch="true" href="/permissions">
														        	<u><a href="">Permissions</a></u>
														        </NavLink>
														    </span>
														</p>
													</>
												    )}

													{/*<p>
											        	<span>Delivery Method:</span>
												      	<span>
													        <NavLink activeClassName='active' prefetch="true" href="/createDeliveryMethod">
													        	<a>Create Delivery</a>
													        </NavLink>
													    </span>
													</p>*/}
										        </>
			      							)}
								    	
									</OrderStyles>
								</div>
							
						);
					}}
					
				</Query>
			</StyledLinked>
		);
	}
}

export default Account;