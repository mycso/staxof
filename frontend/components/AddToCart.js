import React from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { CURRENT_USER_QUERY } from './User';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { 
    faShoppingCart,
} from '@fortawesome/free-solid-svg-icons';

const ADD_TO_CART_MUTATION = gql`
	mutation addToCart($id: ID!) {
		addToCart(id: $id) {
			id
			quantity
		}
	}
`;

class AddToCart extends React.Component {
	render() {
		const { id } = this.props;
		return ( 
			<Mutation 
				mutation={ADD_TO_CART_MUTATION} 
				variables={{
					id,
				}}
				refetchQueries={[{ query: CURRENT_USER_QUERY }]}
			>
				{(addToCart, {loading}) => 
					<a>
						<button disabled={loading} onClick={addToCart} className="cart-btn">Buy{loading && 'ing'} Item <FontAwesomeIcon icon={faShoppingCart} /></button>
					</a> 
				}
			</Mutation>
		);
	}
}
export default AddToCart;
export { ADD_TO_CART_MUTATION };