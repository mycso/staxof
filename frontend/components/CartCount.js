import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

const AnimationStyles = styled.span`
	position: relative;
	.count{
		display: block;
		position: absolute;
		transition: all 0.4s;
		backface-visibility: hidden; 
	}
	/* Initial State of the entered Dot */
	.count-enter {
		transform: scale(4) rotateX(0.5turn);
	}
	.count-enter-active {
		transform: rotateX(0);
	}
	.count-exit {
		top: 0;
		position: absolute;
		transform: rotateX(0);
	}
	.count-exit-active {
		transform: scale(4) rotateX(0.5turn);
	}
`;

const Dot = styled.div`
	background: ${props => props.theme.orange};
	color: white;
	border-radius: 20%;
	padding: 0.2rem;
	line-height: 1rem;
	min-width: auto;
    min-height: auto;
    margin-left: 1rem;
    margin-top: -2.7rem;
    font-weight: 100;
    font-feature-settings: 'tnum';
    font-variant-numeric: tabular-nums;
    float: right;
`;

const CartCount = ({ count }) => 
	<AnimationStyles>
		<TransitionGroup>
			<CSSTransition 
				unmountOnExit
				className="count" 
				classNames="count" 
				key={count} 
				timeout={{ enter: 400, exit: 400 }}
			>
				<Dot>{count}</Dot>
			</CSSTransition>
		</TransitionGroup>
	</AnimationStyles>

	
export default CartCount;