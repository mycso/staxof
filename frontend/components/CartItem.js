import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import formatMoney from '../lib/formatMoney';
import PriceTag from './styles/PriceTag';
import PlusToCart from './PlusToCart';
import MinusFromCart from './MinusFromCart';
import RemoveFromCart from './RemoveFromCart';

const CartItemStyles = styled.li`
	padding: 1rem 0;
	border-bottom: 0.5px solid ${props => props.theme.grey};
	display: grid;
	align-items: center;
	grid-template-columns: auto 1fr auto;
	img {
		margin-right: 10px;
	}
	h3,
	p {
		margin: 0;
	}

	@media (max-width: 1214px) {
    	grid-template-columns: 1fr;
  	}
`;

const CartItem = ({ cartItem }) => { 
	// First Check if that item exists
	if(!cartItem.item) 
	return (
		<CartItemStyles>
			<p>This item is currently out of stock</p>
			<RemoveFromCart id={cartItem.id} />
		</CartItemStyles>
	);

	if(cartItem.item) {
		return ( 

			<CartItemStyles>
				<img width="100" src={cartItem.item.image} alt={cartItem.item.title} />
				<div className="cart-item-details">
					<h3>{cartItem.item.title}</h3>
					<p>
						{formatMoney(cartItem.quantity * cartItem.item.price)}
						{/*{formatMoney(cartItem.quantity * cartItem.item.price + cartItem.item.deliveryCost)}*/}
						{' - '}
						<em>
							{cartItem.quantity}
							&times; {formatMoney(cartItem.item.price)} each
							{/*&times; {formatMoney(cartItem.item.price + cartItem.item.deliveryCost)} each*/}
							{/*<p>Delivery Cost: {formatMoney(cartItem.item.deliveryCost * cartItem.quantity)}</p>*/}
						</em>
					</p>
				</div>
				
				<span>
					<PlusToCart id={cartItem.item.id} />
					<PriceTag>{cartItem.quantity}</PriceTag>
					<MinusFromCart prefetch id={cartItem.item.id} />
					<RemoveFromCart prefetch id={cartItem.id} />
				</span>
			</CartItemStyles>

		);
	}
};

CartItem.propTypes = {
	cartItem: PropTypes.object.isRequired,
};

export default CartItem;