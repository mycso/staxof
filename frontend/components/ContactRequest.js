import React, { Component } from 'react';
import gql from 'graphql-tag';
import { Mutation } from 'react-apollo';
import Form from './styles/Form';
import Error from './ErrorMessage';

const CONTACT_REQUEST_MUTATION = gql `
	mutation CONTACT_REQUEST_MUTATION(
		$email: String!
		$subject: String! 
		$message: String!	
	) {
		createContact(
			email: $email 
			subject: $subject
			message: $message
		) {
			message
		}
	}
`;

class ContactRequest extends Component {
	state = {
		email: '',
		subject: '', 
		message: ''
	};
	saveToState = e => {
		this.setState({ [e.target.name]: e.target.value })
	};
	render() {
		return (
			<Mutation 
				mutation={CONTACT_REQUEST_MUTATION} 
				variables={this.state}
			>
				{(reset, { error, loading, called }) => (
						<Form 
							method="post" 
							onSubmit={async e => {
								e.preventDefault();
								await reset();
								this.setState({ email: '', subject: '', message: '' });
							}}
						>
							<fieldset disabled={loading} aria-busy={loading}>

								<h2>Contact Us</h2>
								<br />
						    	<h4>Leave a message for us right here and weâ€™ll get back to you as soon as we can!</h4>
						    	<br />

								<Error error={error} />
								{!error && !loading && called && <p>Success! Thank you for your message!</p>}
								<label htmlFor="email">
									Email
									<input 
										type="email" 
										name="email" 
										placeholder="Email" 
										value={this.state.email} 
										onChange={this.saveToState}
										required
									/>
								</label>
								<label htmlFor="subject">
									Subject
									<input 
										type="text" 
										name="subject" 
										placeholder="Subject" 
										value={this.state.subject} 
										onChange={this.saveToState}
										required
									/>
								</label>
								<label htmlFor="message">
									Message
									<textarea 
										type="text" 
										name="message" 
										placeholder="Message" 
										value={this.state.message} 
										onChange={this.saveToState}
										required
									/>
								</label>
								<button type="submit">Send Message</button>
							</fieldset>
						</Form>
					)
				}
			</Mutation>
		);
	}
}

export default ContactRequest;
export { CONTACT_REQUEST_MUTATION };