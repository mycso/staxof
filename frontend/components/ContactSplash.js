import Link from 'next/link';
import styled from 'styled-components';

import { ScrollTo } from "react-scroll-to"
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { 
    faArrowDown,
} from '@fortawesome/free-solid-svg-icons';

import { 
    faFacebook,
    faTwitter,
    faInstagram
} from '@fortawesome/free-brands-svg-icons';

import SplashImage from './styles/SplashImageStyles';
import SubHeader from '../components/SubHeader';

const StyledContactSplash = styled.div`

	.adorable-blur {
		background-image: url('static/adorable-blur.jpg');
		background-size: cover;
		width: 100%;
		height: 100%;
		background-position: center;
	}

`;

const ContactSplash = () => (
	<SplashImage>
		<StyledContactSplash>
			<SubHeader />
			<div>
			  <div id="carouselExampleIndicators" className="carousel slide" data-ride="carousel">
			    <ol className="carousel-indicators d-none">
			      <li data-target="#carouselExampleIndicators" data-slide-to="0" className="active"></li>
			      <li data-target="#carouselExampleIndicators" data-slide-to="1"></li>
			      <li data-target="#carouselExampleIndicators" data-slide-to="2"></li>
			    </ol>
			    <div className="social-button-main">
			      <div className="col-xs-12 col-sm-12 col-md-12">
			        <ul className="list-unstyled list-inline social text-center">
			          <li className="list-inline-item"><a href="https://www.facebook.com/staxofstories/" target="_blank"><FontAwesomeIcon icon={faFacebook} /></a></li>
			          <li className="list-inline-item"><a href="#"><FontAwesomeIcon icon={faTwitter} /></a></li>
			          <li className="list-inline-item"><a href="https://www.instagram.com/staxofstories/" target="_blank"><FontAwesomeIcon icon={faInstagram} /></a></li>
			        </ul>
			      </div>
			    </div> 
			    <div className="carousel-inner" role="listbox">
			      <div className="carousel-item active">
			      	<div class="adorable-blur"></div>
			        <div className="carousel-caption heading d-md-block"> 
			          <h2 className="display-4">Contact Us!</h2>
			          <h3 className="display-5">We are just a call away from you.</h3>
			        </div>
			      </div>
			    </div>
			    <a className="carousel-control-prev d-none" href="#carouselExampleIndicators" role="button" data-slide="prev">
			      <span className="carousel-control-prev-icon" aria-hidden="true"></span>
			      <span className="sr-only">Previous</span>
			    </a>
			    <a className="carousel-control-next d-none" href="#carouselExampleIndicators" role="button" data-slide="next">
			      <span className="carousel-control-next-icon" aria-hidden="true"></span>
			      <span className="sr-only">Next</span>
			    </a>
			  </div>
		  	</div>
	  	</StyledContactSplash>
	</SplashImage>
);

export default ContactSplash;