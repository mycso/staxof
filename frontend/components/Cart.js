import React, { Component } from 'react';
// import { useQuery, useApolloClient, useState } from '@apollo/react-hooks';
import { Query, Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { adopt } from 'react-adopt';
import { useUser } from './User';
import styled from 'styled-components';
import User from './User';
import CartStyles from './styles/CartStyles';
import CartTitle from './styles/CartTitle';
import CloseButton from './styles/CloseButton';
import CartItem from './CartItem';
import Form from './styles/Form';
import calcTotalPrice from '../lib/calcTotalPrice';
// import calcTotalDeliveryPrice from '../lib/calcTotalDeliveryPrice';
import formatMoney from '../lib/formatMoney';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { 
    faTimes,
} from '@fortawesome/free-solid-svg-icons';
import { 
    faChevronRight,
} from '@fortawesome/free-solid-svg-icons';
import TakeMyMoney from './TakeMyMoney';

const LOCAL_STATE_QUERY = gql`
	query {
		cartOpen @client
	}
`;

const TOGGLE_CART_MUTATION = gql`
	mutation {
		toggleCart @client
	}
`;

/* eslint-disable */
/* eslint-disable */

const Composed = adopt({
	user: ({ render }) => <User>{render}</User>,
	toggleCart: ({ render }) => <Mutation mutation={TOGGLE_CART_MUTATION}>{render}</Mutation>,
	localState: ({ render }) => <Query query={LOCAL_STATE_QUERY}>{render}</Query>,
});

/* eslint-enable */

// const international = 1100;
// const royalmail = 500;

const Cart = ({ cartItem }) => (

	<Composed>
		{({user, toggleCart, localState}) => {
		const me = user.data.me;
			
			if(!me) return null;

			return (
				<CartStyles open={localState.data.cartOpen}>
					<header>
						<CloseButton onClick={toggleCart} title="close"><FontAwesomeIcon icon={faTimes} /></CloseButton>
						<CartTitle>{me.email}'s Cart</CartTitle>
						<p>You have {me.cart.length} Item{me.cart.length === 1 ? '' : 's'} in your cart</p>
					</header>
					<ul>
						{me.cart.map(cartItem => 
							<CartItem key={cartItem.id} cartItem={cartItem} />
						)}
					</ul>

					<footer>

				    	{/*<div>
				    		Delivery Cost: {formatMoney(calcTotalDeliveryPrice(me.cart))}
				    	</div>*/}

						<div>
							<span>Total: {formatMoney(calcTotalPrice(me.cart))}</span>
						</div>

						<div>
							{me.cart.length > 0 && (
								<TakeMyMoney>
									<button id="buyIt" className="btn btn-success checkout" type="submit">Checkout <FontAwesomeIcon icon={faChevronRight} /></button>
								</TakeMyMoney>
							)}
						</div>
						
					</footer>
				</CartStyles>
			);
		}}
	</Composed>
);

export default Cart;
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION };