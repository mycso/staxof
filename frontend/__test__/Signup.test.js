import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import { ApolloConsumer } from 'react-apollo';
import Signup, { SIGNUP_MUTATION } from '../components/Signup';
import { CURRENT_USER_USER } from '../components/User';
import { fakeUser } from '../lib/testUtils';

function type(wrapper, email, value){
	wrapper.find(`input[name="${email}"]`).simulate('change', {
		target: { email, value },
	});
}

const me = fakeUser();
const mocks = [
	{
		request: {
			query: SIGNUP_MUTATION,
			variables: {
				terms: me.terms,
				password: 'cobra14',
				email: me.email,
				
			},
		},
		result: {
			data: { 
				signup: {
					__typename: 'User',
					id: 'abc123',
					email: me.email,
					terms: me.terms,
				},
			},
		},
	},

	// current user query mock
	{
		request: { query: CURRENT_USER_USER },
		result: { data: { me }},
	},
];

describe('<Signup/>', () => {
 it('renders and matches snapshot', async () => {
 	const wrapper = mount(
 		<MockedProvider>
 			<Signup />
 		</MockedProvider>
 	);
 	expect(toJSON(wrapper.find('form'))).toMatchSnapshot(); 
 });

 it('calls the mutation properly', async () => {
 	let apolloClient;
 	const wrapper = mount(
	 	<MockedProvider mocks={mocks}>
	 		<ApolloConsumer>
	 			{client => {
	 				apolloClient = client;
	 				return <Signup />;
	 			}}
	 		</ApolloConsumer>
	 	</MockedProvider>
	);
	await wait();
	wrapper.update();
	type(wrapper, 'terms', me.terms);
	type(wrapper, 'password', 'cobra14');
	type(wrapper, 'email', me.email);
	wrapper.update();
	wrapper.find('form').simulate('submit');
	await wait();
	// query the user out of the apollo client
	const user = await apolloClient.query({ query: CURRENT_USER_USER });
	expect(user.data.me).toMatchObject(me);
 });
});