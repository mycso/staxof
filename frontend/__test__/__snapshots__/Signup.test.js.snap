// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup/> renders and matches snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 fuiEQo"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Sign Up for an Account
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </label>
    <label
      htmlFor="password"
    >
      Password
      <input
        name="password"
        onChange={[Function]}
        pattern=".{10,}"
        placeholder="Password"
        required={true}
        title="10 or more characters"
        type="password"
        value=""
      />
    </label>
    <Signup__Terms>
      <div
        className="Signup__Terms-sc-1bh7ije-0 eBigCh"
      >
        <br />
        <div
          htmlFor="terms"
        >
          <h6
            className="terms"
          >
            <input
              className="mr-2"
              name="terms"
              onChange={[Function]}
              required={true}
              type="checkbox"
              value={true}
            />
            By signing up, you agree to our 
            <a
              className="links"
              href="/termsConditions"
              target="_blank"
            >
              Terms of Use
            </a>
             and 
            <a
              className="links"
              href="/privacyPolicy"
              target="_blank"
            >
              Privacy Policy
            </a>
            .
          </h6>
        </div>
        <br />
      </div>
    </Signup__Terms>
    <button
      type="submit"
    >
      Sign Up!
    </button>
  </fieldset>
</form>
`;
